poetry/inspection/info.py

```diff
@@ -392,10 +401,44 @@ def _get_poetry_package(path):  # type: (Path) -> Optional[ProjectPackage]
         except RuntimeError:
             pass
 
+    @classmethod
+    def _pep517_metadata(cls, path):  # type (Path) -> PackageInfo
+        """
+        Helper method to use PEP-517 library to build and read package metadata.
+
+        :param path: Path to package source to build and read metadata for.
+        """
+        info = None
+        try:
+            info = cls.from_setup_files(path)
+            if info.requires_dist is not None:
+                return info
+        except PackageInfoError:
+            pass
+
+        try:
+            with temporary_directory() as tmp_dir:
+                pep517_metadata_build(
```

Comment:
The issue I have with this is that `pep517` uses `sys.executable` (i.e. the Python interpreter used to run Poetry) and, thus, is not compatible with Poetry's environment system.

That's why I mentioned in https://github.com/python-poetry/poetry/pull/2296#issuecomment-636839053 that we would most likely not be able to use `pep517`.

Reply:
We can workaround that by executing `python -m pep517.meta.build` under the project's environment.

Reply:
The python version of Poetry's environment is not necessarily the same as the one from the project's environment so if a build backend relies on it, and this is particularly true for setuptools, we might end up with the wrong information.

Reply:
@sdispater I have added a commit wherein we retail the old behaviour wrt venv before building metadata.